vavr/src/main/java/io/vavr/control/Validation.java

```diff
@@ -739,6 +739,19 @@ default void forEach(Consumer<? super T> action) {
         return this;
     }
 
+    /**
+     * Consumes the errors if this is an Invalid.
+     *
+     * @param action The action that will be performed on the errors.
+     * @return this instance
+     */
+    default Validation<E, T> peekInvalid(Consumer<? super Seq<E>> action) {
+        if (isInvalid()) {
+            action.accept(getErrors());
+        }
+        return this;
+    }
```

Comment:
I tested these:

```java
    default void test() {
        Validation<String, String> v = Validation.valid("A");

        Consumer<Seq<String>> action1 = System.out::println;
        v.peekInvalid(action1); // OK

        Consumer<Seq<CharSequence>> action2 = System.out::println;
        v.peekInvalid(action2); // TYPE ERROR

        Consumer<Iterable<String>> action3 = System.out::println;
        v.peekInvalid(action3); // OK

        Consumer<Iterable<CharSequence>> action4 = System.out::println;
        v.peekInvalid(action4); // TYPE ERROR
    }
```

However, I think your type signature is correct! Using `Consumer<? super Seq<? super E>> action` does not work.